marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -25,6 +25,11 @@ object Api {
 
   val versionString = "v2"
 
+  case class Auth(
```

Comment:
What about passing in a uid and secret instead of content and letting the Api implementation construct the json content body?